### ISSUE TITLE
Feature: WASM/emscripten support

### ISSUE THREAD

**Issue by Skycoder42** [weidai11/cryptopp#647]
With the WebAssembly (http://webassembly.org/) standard now officially released and support in all major browsers, it would be really awesome if we could officially compile cryptopp with the emscripten SDK to create wasm binaries.

Im no expert on the matter, but it should not be any more complicated that supporting any new compiler - leave out any architecture optimizations and it (should) work?

**Comment by Skycoder42** [comment]
So I just tried to compile it by settings the AR/CXX/CC environment variables and surprisignly enough it complies with `make -f GNUMakefile static test` on my system. Obviously we are not done yet, as compiler flags etc need to be worked out, but at least cryptopp generally compiles against emscripten!

I am currently trying to figure out how to run the genrated cryptest.exe in my browser. I will post an update as soon as I manage to do that.

**Comment by Skycoder42** [comment]
So after playing around a little I manage to get the test to execute. It fails with a file access exception, as file access with wasm is simply not that simple - but at least it does execute!

I can post the changes I made to the makefile - but be warned! I simpyl copied what the Qt-Project is doing to find out if it is theoretically possible to compile with wasm - I did not check all the options etc. so my simple playing around is most likely not suited for "production". But if you are interested I can still add it.
